blueman/plugins/mechanism/Rfcomm.py

```diff
@@ -22,3 +23,6 @@ def close_rfcomm(self, port_id):
         for line in out.decode("UTF-8").splitlines():
             if command in line:
                 os.kill(int(line.split(None, 1)[0]), signal.SIGTERM)
+
+        # FIXME for some reason only root can release, revisit in the future
+        release_rfcomm_device(port_id)
```

Comment:
Is there another process keeping the device open. ModemManager comes up in web searches.

Reply:
Also, what happens if try to release it by command line? Permission denied or something else?

Reply:
OK, did some more tests. There *is* a difference between root and non-privileged (but in dialout group).

Steps:
1. Open a python terminal
2. Call `create_rfcomm_device`
3. Call `release_rfcomm_device`
4. Call `create_rfcomm_device`
5. Run `blueman-rfcomm-watcher /dev/rfcomm0`
6. Call `release_rfcomm_device`
7. Call `create_rfcomm_device`
8. Run `blueman-rfcomm-watcher /dev/rfcomm0`, then kill it -> `/dev/rfcomm0` gets removed
9. Call `release_rfcomm_device`

Step 3 always succeeds. Step 9 always gives error 13 as the device is gone. If I run the python terminal as an unprivileged user in the dialout group, step 6 also gives error 13. If I run it as root, step 6 succeeds. So, it looks like, other than an unprivileged user, root can release a device that is currently in use.

However, we should have killed `blueman-rfcomm-watcher` at this point and there's probably no point in releasing the device if anything else is still using it as it will get auto-closed after that.

Reply:
The problem is that if we are disconnecting the device there is no way other users of the device can keep using it. What I'll do is add a ``rfcomm_release`` method on Mechanism and implement the logic to deal with disconnecting the device and disconnecting the service.